File: Editorial/136_Single_Number/Go/2_Hash_Table.go
func singleNumber(nums []int) int {
    hash_table := make(map[int]int)
    for _, num := range nums {
        hash_table[num]++
    }
    for _, num := range nums {
        if hash_table[num] == 1 {
            return num
        }
    }
    return 0
}

File: Editorial/136_Single_Number/TypeScript/2_Hash_Table.ts
function singleNumber(nums: number[]): number {
    let hash_table: { [key: number]: number } = {};
    for (let num of nums) {
        if (hash_table[num]) {
            hash_table[num]++;
        } else {
            hash_table[num] = 1;
        }
    }
    for (let num of nums) {
        if (hash_table[num] == 1) {
            return num;
        }
    }
    return 0;
}
